Workaround for https://llvm.org/bugs/show_bug.cgi?id=25565
Parallel builds (e.g. `make -j4`) would occasionally fail due to missing
dependencies in the makefiles.

diff --git a/include/llvm/IR/CMakeLists.txt b/include/llvm/IR/CMakeLists.txt
index eade87e..851c4fd 100644
--- a/include/llvm/IR/CMakeLists.txt
+++ b/include/llvm/IR/CMakeLists.txt
@@ -1,5 +1,6 @@
 set(LLVM_TARGET_DEFINITIONS Attributes.td)
 tablegen(LLVM Attributes.inc -gen-attrs)
+add_public_tablegen_target(attributes_inc)
 
 set(LLVM_TARGET_DEFINITIONS Intrinsics.td)
 tablegen(LLVM Intrinsics.gen -gen-intrinsic)
diff --git a/lib/Analysis/CMakeLists.txt b/lib/Analysis/CMakeLists.txt
index 6962361..cd8e1d8 100644
--- a/lib/Analysis/CMakeLists.txt
+++ b/lib/Analysis/CMakeLists.txt
@@ -74,4 +74,4 @@ add_llvm_library(LLVMAnalysis
   ${LLVM_MAIN_INCLUDE_DIR}/llvm/Analysis
   )
 
-add_dependencies(LLVMAnalysis intrinsics_gen)
+add_dependencies(LLVMAnalysis intrinsics_gen attributes_inc)
diff --git a/lib/AsmParser/CMakeLists.txt b/lib/AsmParser/CMakeLists.txt
index 0d72723..a32215b 100644
--- a/lib/AsmParser/CMakeLists.txt
+++ b/lib/AsmParser/CMakeLists.txt
@@ -9,4 +9,5 @@ add_llvm_library(LLVMAsmParser
 
   DEPENDS
   intrinsics_gen
+  attributes_inc
   )
diff --git a/lib/Bitcode/Reader/CMakeLists.txt b/lib/Bitcode/Reader/CMakeLists.txt
index 62954f2..63e3ced 100644
--- a/lib/Bitcode/Reader/CMakeLists.txt
+++ b/lib/Bitcode/Reader/CMakeLists.txt
@@ -7,4 +7,4 @@ add_llvm_library(LLVMBitReader
   ${LLVM_MAIN_INCLUDE_DIR}/llvm/Bitcode
   )
 
-add_dependencies(LLVMBitReader intrinsics_gen)
+add_dependencies(LLVMBitReader intrinsics_gen attributes_inc)
diff --git a/lib/Bitcode/Writer/CMakeLists.txt b/lib/Bitcode/Writer/CMakeLists.txt
index 82dc6b2..41d3c52 100644
--- a/lib/Bitcode/Writer/CMakeLists.txt
+++ b/lib/Bitcode/Writer/CMakeLists.txt
@@ -6,4 +6,5 @@ add_llvm_library(LLVMBitWriter
 
   DEPENDS
   intrinsics_gen
+  attributes_inc
   )
diff --git a/lib/CodeGen/CMakeLists.txt b/lib/CodeGen/CMakeLists.txt
index a078c3c..ddb421f 100644
--- a/lib/CodeGen/CMakeLists.txt
+++ b/lib/CodeGen/CMakeLists.txt
@@ -140,7 +140,7 @@ add_llvm_library(LLVMCodeGen
   LINK_LIBS ${system_libs}
   )
 
-add_dependencies(LLVMCodeGen intrinsics_gen)
+add_dependencies(LLVMCodeGen intrinsics_gen attributes_inc)
 
 add_subdirectory(SelectionDAG)
 add_subdirectory(AsmPrinter)
diff --git a/lib/CodeGen/SelectionDAG/CMakeLists.txt b/lib/CodeGen/SelectionDAG/CMakeLists.txt
index fbedf2c..b6a72a0 100644
--- a/lib/CodeGen/SelectionDAG/CMakeLists.txt
+++ b/lib/CodeGen/SelectionDAG/CMakeLists.txt
@@ -25,4 +25,4 @@ add_llvm_library(LLVMSelectionDAG
   TargetSelectionDAGInfo.cpp
   )
 
-add_dependencies(LLVMSelectionDAG intrinsics_gen)
+add_dependencies(LLVMSelectionDAG intrinsics_gen attributes_inc)
diff --git a/lib/ExecutionEngine/CMakeLists.txt b/lib/ExecutionEngine/CMakeLists.txt
index 2d9337b..5fd9752 100644
--- a/lib/ExecutionEngine/CMakeLists.txt
+++ b/lib/ExecutionEngine/CMakeLists.txt
@@ -12,6 +12,7 @@ add_llvm_library(LLVMExecutionEngine
 
   DEPENDS
   intrinsics_gen
+  attributes_inc
   )
 
 add_subdirectory(Interpreter)
diff --git a/lib/ExecutionEngine/Interpreter/CMakeLists.txt b/lib/ExecutionEngine/Interpreter/CMakeLists.txt
index 4dbc2df..98a6a75 100644
--- a/lib/ExecutionEngine/Interpreter/CMakeLists.txt
+++ b/lib/ExecutionEngine/Interpreter/CMakeLists.txt
@@ -16,4 +16,4 @@ if( LLVM_ENABLE_FFI )
   target_link_libraries( LLVMInterpreter PRIVATE ${FFI_LIBRARY_PATH} )
 endif()
 
-add_dependencies(LLVMInterpreter intrinsics_gen)
+add_dependencies(LLVMInterpreter intrinsics_gen attributes_inc)
diff --git a/lib/ExecutionEngine/MCJIT/CMakeLists.txt b/lib/ExecutionEngine/MCJIT/CMakeLists.txt
index b1e2bc3..963b7d6 100644
--- a/lib/ExecutionEngine/MCJIT/CMakeLists.txt
+++ b/lib/ExecutionEngine/MCJIT/CMakeLists.txt
@@ -3,4 +3,5 @@ add_llvm_library(LLVMMCJIT
 
   DEPENDS
   intrinsics_gen
+  attributes_inc
   )
diff --git a/lib/ExecutionEngine/OProfileJIT/CMakeLists.txt b/lib/ExecutionEngine/OProfileJIT/CMakeLists.txt
index d585136..b9feec7 100644
--- a/lib/ExecutionEngine/OProfileJIT/CMakeLists.txt
+++ b/lib/ExecutionEngine/OProfileJIT/CMakeLists.txt
@@ -5,3 +5,5 @@ add_llvm_library(LLVMOProfileJIT
   OProfileJITEventListener.cpp
   OProfileWrapper.cpp
   )
+
+add_dependencies(LLVMOProfileJIT attributes_inc)
diff --git a/lib/ExecutionEngine/Orc/CMakeLists.txt b/lib/ExecutionEngine/Orc/CMakeLists.txt
index d26f212..69e1a45 100644
--- a/lib/ExecutionEngine/Orc/CMakeLists.txt
+++ b/lib/ExecutionEngine/Orc/CMakeLists.txt
@@ -14,4 +14,5 @@ add_llvm_library(LLVMOrcJIT
 
   DEPENDS
   intrinsics_gen
+  attributes_inc
   )
diff --git a/lib/ExecutionEngine/RuntimeDyld/CMakeLists.txt b/lib/ExecutionEngine/RuntimeDyld/CMakeLists.txt
index 182f982..d503f3b 100644
--- a/lib/ExecutionEngine/RuntimeDyld/CMakeLists.txt
+++ b/lib/ExecutionEngine/RuntimeDyld/CMakeLists.txt
@@ -8,4 +8,5 @@ add_llvm_library(LLVMRuntimeDyld
 
   DEPENDS
   intrinsics_gen
+  attributes_inc
   )
diff --git a/lib/IR/CMakeLists.txt b/lib/IR/CMakeLists.txt
index 40b4ec6..3faa9f2 100644
--- a/lib/IR/CMakeLists.txt
+++ b/lib/IR/CMakeLists.txt
@@ -64,4 +64,4 @@ if( MSVC_VERSION EQUAL 1800)
     )
 endif()
 
-add_dependencies(LLVMCore intrinsics_gen)
+add_dependencies(LLVMCore intrinsics_gen attributes_inc)
diff --git a/lib/IRReader/CMakeLists.txt b/lib/IRReader/CMakeLists.txt
index 87ea880..66ec484 100644
--- a/lib/IRReader/CMakeLists.txt
+++ b/lib/IRReader/CMakeLists.txt
@@ -6,4 +6,5 @@ add_llvm_library(LLVMIRReader
 
   DEPENDS
   intrinsics_gen
+  attributes_inc
   )
diff --git a/lib/LTO/CMakeLists.txt b/lib/LTO/CMakeLists.txt
index 1c099bb..4a445b9 100644
--- a/lib/LTO/CMakeLists.txt
+++ b/lib/LTO/CMakeLists.txt
@@ -6,4 +6,4 @@ add_llvm_library(LLVMLTO
   ${LLVM_MAIN_INCLUDE_DIR}/llvm/LTO
   )
 
-add_dependencies(LLVMLTO intrinsics_gen)
+add_dependencies(LLVMLTO intrinsics_gen attributes_inc)
diff --git a/lib/Linker/CMakeLists.txt b/lib/Linker/CMakeLists.txt
index 8916fb3..2f8a97f 100644
--- a/lib/Linker/CMakeLists.txt
+++ b/lib/Linker/CMakeLists.txt
@@ -7,4 +7,5 @@ add_llvm_library(LLVMLinker
 
   DEPENDS
   intrinsics_gen
+  attributes_inc
   )
diff --git a/lib/Object/CMakeLists.txt b/lib/Object/CMakeLists.txt
index 2ac2ee5..eb6ffc5 100644
--- a/lib/Object/CMakeLists.txt
+++ b/lib/Object/CMakeLists.txt
@@ -23,4 +23,5 @@ add_llvm_library(LLVMObject
 
   DEPENDS
   intrinsics_gen
+  attributes_inc
   )
diff --git a/lib/Passes/CMakeLists.txt b/lib/Passes/CMakeLists.txt
index 6ceac7b..e48db8f 100644
--- a/lib/Passes/CMakeLists.txt
+++ b/lib/Passes/CMakeLists.txt
@@ -5,4 +5,4 @@ add_llvm_library(LLVMPasses
   ${LLVM_MAIN_INCLUDE_DIR}/llvm/Passes
   )
 
-add_dependencies(LLVMPasses intrinsics_gen)
+add_dependencies(LLVMPasses intrinsics_gen attributes_inc)
diff --git a/lib/ProfileData/CMakeLists.txt b/lib/ProfileData/CMakeLists.txt
index 22cca4b..4d181d4 100644
--- a/lib/ProfileData/CMakeLists.txt
+++ b/lib/ProfileData/CMakeLists.txt
@@ -14,4 +14,5 @@ add_llvm_library(LLVMProfileData
 
   DEPENDS
   intrinsics_gen
+  attributes_inc
   )
diff --git a/lib/Target/AArch64/CMakeLists.txt b/lib/Target/AArch64/CMakeLists.txt
index f26327f..b30b452 100644
--- a/lib/Target/AArch64/CMakeLists.txt
+++ b/lib/Target/AArch64/CMakeLists.txt
@@ -45,7 +45,7 @@ add_llvm_target(AArch64CodeGen
   AArch64TargetTransformInfo.cpp
 )
 
-add_dependencies(LLVMAArch64CodeGen intrinsics_gen)
+add_dependencies(LLVMAArch64CodeGen intrinsics_gen attributes_inc)
 
 add_subdirectory(TargetInfo)
 add_subdirectory(AsmParser)
diff --git a/lib/Target/AMDGPU/CMakeLists.txt b/lib/Target/AMDGPU/CMakeLists.txt
index b9ef0e8..9d86c7d 100644
--- a/lib/Target/AMDGPU/CMakeLists.txt
+++ b/lib/Target/AMDGPU/CMakeLists.txt
@@ -69,3 +69,5 @@ add_subdirectory(InstPrinter)
 add_subdirectory(TargetInfo)
 add_subdirectory(MCTargetDesc)
 add_subdirectory(Utils)
+
+add_dependencies(LLVMAMDGPUCodeGen attributes_inc)
diff --git a/lib/Target/ARM/CMakeLists.txt b/lib/Target/ARM/CMakeLists.txt
index 0b698197..6012073 100644
--- a/lib/Target/ARM/CMakeLists.txt
+++ b/lib/Target/ARM/CMakeLists.txt
@@ -51,3 +51,5 @@ add_subdirectory(AsmParser)
 add_subdirectory(Disassembler)
 add_subdirectory(InstPrinter)
 add_subdirectory(MCTargetDesc)
+
+add_dependencies(LLVMARMCodeGen attributes_inc)
diff --git a/lib/Target/AVR/CMakeLists.txt b/lib/Target/AVR/CMakeLists.txt
index b4fb0d9..d5cce72 100644
--- a/lib/Target/AVR/CMakeLists.txt
+++ b/lib/Target/AVR/CMakeLists.txt
@@ -9,7 +9,7 @@ add_llvm_target(AVRCodeGen
     AVRTargetObjectFile.cpp
   )
 
-add_dependencies(LLVMAVRCodeGen intrinsics_gen)
+add_dependencies(LLVMAVRCodeGen intrinsics_gen attributes_inc)
 
 add_subdirectory(TargetInfo)
 
diff --git a/lib/Target/CMakeLists.txt b/lib/Target/CMakeLists.txt
index e6d0199..4e32b13 100644
--- a/lib/Target/CMakeLists.txt
+++ b/lib/Target/CMakeLists.txt
@@ -17,3 +17,5 @@ foreach(t ${LLVM_TARGETS_TO_BUILD})
   message(STATUS "Targeting ${t}")
   add_subdirectory(${t})
 endforeach()
+
+add_dependencies(LLVMTarget attributes_inc)
diff --git a/lib/Target/MSP430/CMakeLists.txt b/lib/Target/MSP430/CMakeLists.txt
index 3f37763..62e9e1b 100644
--- a/lib/Target/MSP430/CMakeLists.txt
+++ b/lib/Target/MSP430/CMakeLists.txt
@@ -25,3 +25,5 @@ add_llvm_target(MSP430CodeGen
 add_subdirectory(InstPrinter)
 add_subdirectory(TargetInfo)
 add_subdirectory(MCTargetDesc)
+
+add_dependencies(LLVMMSP430CodeGen attributes_inc)
diff --git a/lib/Target/Mips/CMakeLists.txt b/lib/Target/Mips/CMakeLists.txt
index bde843a..e8565bf 100644
--- a/lib/Target/Mips/CMakeLists.txt
+++ b/lib/Target/Mips/CMakeLists.txt
@@ -54,3 +54,4 @@ add_subdirectory(TargetInfo)
 add_subdirectory(MCTargetDesc)
 add_subdirectory(AsmParser)
 
+add_dependencies(LLVMMipsCodeGen attributes_inc)
diff --git a/lib/Target/NVPTX/CMakeLists.txt b/lib/Target/NVPTX/CMakeLists.txt
index 05fe06d..846d014 100644
--- a/lib/Target/NVPTX/CMakeLists.txt
+++ b/lib/Target/NVPTX/CMakeLists.txt
@@ -39,3 +39,5 @@ add_llvm_target(NVPTXCodeGen ${NVPTXCodeGen_sources})
 add_subdirectory(TargetInfo)
 add_subdirectory(InstPrinter)
 add_subdirectory(MCTargetDesc)
+
+add_dependencies(LLVMNVPTXCodeGen attributes_inc)
diff --git a/lib/Target/PowerPC/CMakeLists.txt b/lib/Target/PowerPC/CMakeLists.txt
index c31abab..c0999c4 100644
--- a/lib/Target/PowerPC/CMakeLists.txt
+++ b/lib/Target/PowerPC/CMakeLists.txt
@@ -46,3 +46,5 @@ add_subdirectory(Disassembler)
 add_subdirectory(InstPrinter)
 add_subdirectory(TargetInfo)
 add_subdirectory(MCTargetDesc)
+
+add_dependencies(LLVMPowerPCCodeGen attributes_inc)
diff --git a/lib/Target/Sparc/CMakeLists.txt b/lib/Target/Sparc/CMakeLists.txt
index 5b7bfdd..b2e5941 100644
--- a/lib/Target/Sparc/CMakeLists.txt
+++ b/lib/Target/Sparc/CMakeLists.txt
@@ -31,3 +31,5 @@ add_subdirectory(MCTargetDesc)
 add_subdirectory(InstPrinter)
 add_subdirectory(AsmParser)
 add_subdirectory(Disassembler)
+
+add_dependencies(LLVMSparcCodeGen attributes_inc)
diff --git a/lib/Target/SystemZ/CMakeLists.txt b/lib/Target/SystemZ/CMakeLists.txt
index 336f037..fad2137 100644
--- a/lib/Target/SystemZ/CMakeLists.txt
+++ b/lib/Target/SystemZ/CMakeLists.txt
@@ -37,3 +37,5 @@ add_subdirectory(Disassembler)
 add_subdirectory(InstPrinter)
 add_subdirectory(TargetInfo)
 add_subdirectory(MCTargetDesc)
+
+add_dependencies(LLVMSystemZCodeGen attributes_inc)
diff --git a/lib/Target/WebAssembly/CMakeLists.txt b/lib/Target/WebAssembly/CMakeLists.txt
index e5c68e5..ff81bc3 100644
--- a/lib/Target/WebAssembly/CMakeLists.txt
+++ b/lib/Target/WebAssembly/CMakeLists.txt
@@ -37,7 +37,7 @@ add_llvm_target(WebAssemblyCodeGen
   WebAssemblyTargetTransformInfo.cpp
 )
 
-add_dependencies(LLVMWebAssemblyCodeGen intrinsics_gen)
+add_dependencies(LLVMWebAssemblyCodeGen intrinsics_gen attributes_inc)
 
 add_subdirectory(Disassembler)
 add_subdirectory(InstPrinter)
diff --git a/lib/Target/X86/CMakeLists.txt b/lib/Target/X86/CMakeLists.txt
index 5594915..946faca 100644
--- a/lib/Target/X86/CMakeLists.txt
+++ b/lib/Target/X86/CMakeLists.txt
@@ -46,3 +46,5 @@ add_subdirectory(InstPrinter)
 add_subdirectory(MCTargetDesc)
 add_subdirectory(TargetInfo)
 add_subdirectory(Utils)
+
+add_dependencies(LLVMX86CodeGen attributes_inc)
diff --git a/lib/Target/XCore/CMakeLists.txt b/lib/Target/XCore/CMakeLists.txt
index 0a609ef..677bd43 100644
--- a/lib/Target/XCore/CMakeLists.txt
+++ b/lib/Target/XCore/CMakeLists.txt
@@ -30,3 +30,5 @@ add_subdirectory(Disassembler)
 add_subdirectory(InstPrinter)
 add_subdirectory(TargetInfo)
 add_subdirectory(MCTargetDesc)
+
+add_dependencies(LLVMXCoreCodeGen attributes_inc)
diff --git a/lib/Transforms/IPO/CMakeLists.txt b/lib/Transforms/IPO/CMakeLists.txt
index 351b88f..df0fc8a 100644
--- a/lib/Transforms/IPO/CMakeLists.txt
+++ b/lib/Transforms/IPO/CMakeLists.txt
@@ -33,4 +33,4 @@ add_llvm_library(LLVMipo
   ${LLVM_MAIN_INCLUDE_DIR}/llvm/Transforms/IPO
   )
 
-add_dependencies(LLVMipo intrinsics_gen)
+add_dependencies(LLVMipo intrinsics_gen attributes_inc)
diff --git a/lib/Transforms/InstCombine/CMakeLists.txt b/lib/Transforms/InstCombine/CMakeLists.txt
index 0ed8e62..e104ae9 100644
--- a/lib/Transforms/InstCombine/CMakeLists.txt
+++ b/lib/Transforms/InstCombine/CMakeLists.txt
@@ -18,4 +18,4 @@ add_llvm_library(LLVMInstCombine
   ${LLVM_MAIN_INCLUDE_DIR}/llvm/Transforms/InstCombine
   )
 
-add_dependencies(LLVMInstCombine intrinsics_gen)
+add_dependencies(LLVMInstCombine intrinsics_gen attributes_inc)
diff --git a/lib/Transforms/Instrumentation/CMakeLists.txt b/lib/Transforms/Instrumentation/CMakeLists.txt
index cae1e5a..cd310d5 100644
--- a/lib/Transforms/Instrumentation/CMakeLists.txt
+++ b/lib/Transforms/Instrumentation/CMakeLists.txt
@@ -15,4 +15,4 @@ add_llvm_library(LLVMInstrumentation
   ${LLVM_MAIN_INCLUDE_DIR}/llvm/Transforms
   )
 
-add_dependencies(LLVMInstrumentation intrinsics_gen)
+add_dependencies(LLVMInstrumentation intrinsics_gen attributes_inc)
diff --git a/lib/Transforms/ObjCARC/CMakeLists.txt b/lib/Transforms/ObjCARC/CMakeLists.txt
index 98ad37f..02c21d2 100644
--- a/lib/Transforms/ObjCARC/CMakeLists.txt
+++ b/lib/Transforms/ObjCARC/CMakeLists.txt
@@ -13,4 +13,4 @@ add_llvm_library(LLVMObjCARCOpts
   ${LLVM_MAIN_INCLUDE_DIR}/llvm/Transforms
   )
 
-add_dependencies(LLVMObjCARCOpts intrinsics_gen)
+add_dependencies(LLVMObjCARCOpts intrinsics_gen attributes_inc)
diff --git a/lib/Transforms/Scalar/CMakeLists.txt b/lib/Transforms/Scalar/CMakeLists.txt
index a0ddbd0..b6f25d3 100644
--- a/lib/Transforms/Scalar/CMakeLists.txt
+++ b/lib/Transforms/Scalar/CMakeLists.txt
@@ -55,4 +55,4 @@ add_llvm_library(LLVMScalarOpts
   ${LLVM_MAIN_INCLUDE_DIR}/llvm/Transforms/Scalar
   )
 
-add_dependencies(LLVMScalarOpts intrinsics_gen)
+add_dependencies(LLVMScalarOpts intrinsics_gen attributes_inc)
diff --git a/lib/Transforms/Utils/CMakeLists.txt b/lib/Transforms/Utils/CMakeLists.txt
index 8308a9b..02853a5 100644
--- a/lib/Transforms/Utils/CMakeLists.txt
+++ b/lib/Transforms/Utils/CMakeLists.txt
@@ -45,4 +45,4 @@ add_llvm_library(LLVMTransformUtils
   ${LLVM_MAIN_INCLUDE_DIR}/llvm/Transforms/Utils
   )
 
-add_dependencies(LLVMTransformUtils intrinsics_gen)
+add_dependencies(LLVMTransformUtils intrinsics_gen attributes_inc)
diff --git a/lib/Transforms/Vectorize/CMakeLists.txt b/lib/Transforms/Vectorize/CMakeLists.txt
index 905c069..7b22d53 100644
--- a/lib/Transforms/Vectorize/CMakeLists.txt
+++ b/lib/Transforms/Vectorize/CMakeLists.txt
@@ -8,4 +8,4 @@ add_llvm_library(LLVMVectorize
   ${LLVM_MAIN_INCLUDE_DIR}/llvm/Transforms
   )
 
-add_dependencies(LLVMVectorize intrinsics_gen)
+add_dependencies(LLVMVectorize intrinsics_gen attributes_inc)
